cylc/flow/host_select.py

```diff
@@ -0,0 +1,532 @@
+"""Functionality for selecting a host from pre-defined list."""
+import ast
+from collections import namedtuple
+from functools import lru_cache
+from io import BytesIO
+from itertools import dropwhile
+import json
+import random
+from time import sleep
+import token
+from tokenize import tokenize
+
+from cylc.flow import LOG
+from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
+from cylc.flow.exceptions import HostSelectException
+from cylc.flow.hostuserutil import get_fqdn_by_host, is_remote_host
+from cylc.flow.remote import remote_cylc_cmd, run_cmd
+
+
+def select_suite_host(cached=True):
```

Comment:
Is this extra layer of function necessary? It's only adding 1 arg and 1 line to the wrapped function.

Reply:
The purpose of this method is to extract the required configurations from the global conf so that the interface is independent of the configuration file. For example when I rename the `thresholds` configuration I will only have to update the code in one place.